poll/poll.py

```diff
@@ -784,6 +784,35 @@ def prepare_data(self):
                 ]
         return [header_row] + data.values()
 
+    def generate_report_data(self, user_state_iterator, limit_responses=None):
```

Comment:
For both blocks, you should also take into account ``limit_responses``.

Reply:
Added a check for `limit_responses`.